File: docs/snippets/Serialization/isa-json/investigation-roundtrip.ts
// docs:begin
import { ArcInvestigation, JsonController } from "@nfdi4plants/arctrl";

const investigation = new ArcInvestigation(
  "Investigation-ISAJSON",
  "ISA-JSON example",
);

const isaJson =
  JsonController.Investigation.toISAJsonString(investigation, 2);

const investigationAgain =
  JsonController.Investigation.fromISAJsonString(isaJson);
// docs:end

// docs:assert
if (investigationAgain.Identifier !== "Investigation-ISAJSON") {
  throw new Error(
    `Expected identifier to survive ISA-JSON roundtrip, got ${investigationAgain.Identifier}`,
  );
}
// docs:endassert

File: docs/snippets/Serialization/isa-json/investigation-roundtrip.py
# docs:begin
from arctrl import ArcInvestigation, JsonController

investigation = ArcInvestigation(
    "Investigation-ISAJSON",
    "ISA-JSON example",
)

isa_json = JsonController.Investigation().to_isajson_string(investigation, 2)

investigation_again = JsonController.Investigation().from_isajson_string(isa_json)
# docs:end

# docs:assert
if investigation_again.Identifier != "Investigation-ISAJSON":
    raise Exception(
        "Expected identifier to survive ISA-JSON roundtrip, "
        f"got {investigation_again.Identifier}"
    )
# docs:endassert
